opensciencegrid/contact-sync/topology_utils.py

```diff
@@ -0,0 +1,375 @@
+            service = service.strip().lower()
+            service_id = SERVICE_IDS.get(service)
+            if not service_id:
+                raise Exception("Requested service %s not known; known service"
+                                " names: %s" % (service, ", ".join(SERVICE_IDS)))
+            qs_list.append(("service_sel[]", str(service_id)))
+
+    if getattr(args, 'owner_vo', None):
+        vo_map = get_vo_map(args, session)
+        if 'voown' not in qs_dict:
+            qs_list.append(("voown", "on"))
+        for vo in args.owner_vo.split(","):
+            vo = vo.strip().lower()
+            vo_id = vo_map.get(vo)
+            if not vo_id:
+                raise Exception("Requested owner VO %s not known; known VOs: %s" \
+                    % (vo, ", ".join(vo_map)))
+            qs_list.append(("voown_sel[]", str(vo_id)))
+
+    #url_list[3] = urllib.urlencode(qs_list, doseq=True)
+    url_list[3] = urlparse.urlencode(qs_list, doseq=True)
+
+    return urlparse.urlunsplit(url_list)
+
+
+def get_contacts(args, urltype, roottype):
+    """
+    Get one type of contacts for OSG.
+    """
+    old_no_proxy = os.environ.pop('no_proxy', None)
+    os.environ['no_proxy'] = '.opensciencegrid.org'
+
+    base_url = "https://my.opensciencegrid.org/" + urltype + "summary/xml?" \
+               "&active=on&active_value=1&disable=on&disable_value=0"
+    with get_auth_session(args) as session:
+        url = mangle_url(base_url, args, session)
+        #print(url)
+        response = session.get(url)
+
+    if old_no_proxy is not None:
+        os.environ['no_proxy'] = old_no_proxy
+    else:
+        del os.environ['no_proxy']
+
+    if response.status_code != requests.codes.ok:
+        print("MyOSG request failed (status %d): %s" % \
+              (response.status_code, response.text[:2048]), file=sys.stderr)
+        return None
+
+    root = ET.fromstring(response.content)
+    if root.tag != roottype + 'Summary':
+        print("MyOSG returned invalid XML with root tag %s" % root.tag,
+              file=sys.stderr)
+        return None
+
+    return root
+
+
+def get_vo_contacts(args):
+    """
+    Get resource contacts for OSG.  Return results.
+    """
+    root = get_contacts(args, 'vo', 'VO')
+    if root is None:
+        return 1
+
+    results = {}
+    for child_vo in root:
+        if child_vo.tag != "VO":
+            print("MyOSG returned a non-VO (%s) inside summary." % \
+                  root.tag, file=sys.stderr)
+            return 1
+        name = None
+        contact_list_info = []
+        for item in child_vo:
+            if item.tag == 'Name':
+                name = item.text
+            if item.tag == "ContactTypes":
+                for contact_type in item:
+                    contact_list_info.extend( \
+                        get_contact_list_info(contact_type))
+
+        if name and contact_list_info:
+            results[name] = contact_list_info
+
+    return results
```

Comment:
So you are returning `1` on error otherwise a dict?

That seems like an unusual convention.

Would it make more sense to return `None` on errors?  (If you actually want to return a different type than a dict.)  Or even an exception.